src/machine/arithmetic_ops.rs

```diff
@@ -1067,16 +1067,20 @@ pub(crate) fn truncate(n: Number, arena: &mut Arena) -> Number {
     }
 }
 
-pub(crate) fn round(n: Number, arena: &mut Arena) -> Result<Number, MachineStubGen> {
-    let stub_gen = || {
-        let is_atom = atom!("is");
-        functor_stub(is_atom, 2)
+pub(crate) fn round(num: Number, arena: &mut Arena) -> Result<Number, MachineStubGen> {
+    let res = match num {
+        Number::Fixnum(_) | Number::Integer(_) => {
+            num
+        },
+        Number::Rational(rat) => {
+            Number::arena_from(rat.round(), arena)
```

Comment:
This will create a bigint if `rat` is small enough, but `rnd_i` takes care of reifying it later in this function.